### PR TITLE
refactor(examples) Update federated kaplan meier fitter

### PR DESCRIPTION
**Pull request by adam-narozniak**
## Issue
The example does not use the new Flower API.

## Proposal
Update the example to follow the structure and code of the new Flower.

### CODE REVIEW

File: examples/federated-kaplan-meier-fitter/README.md [hunk 1]
```diff
@@ -25,86 +25,63 @@ the group it comes from therefore to simulate the division that might occur.
...
+└── README.md
 ```
 
-### Installing Dependencies
-
-Project dependencies (such as `lifelines` and `flwr`) are defined in `pyproject.toml` and `requirements.txt`. We recommend [Poetry](https://python-poetry.org/docs/) to install those dependencies and manage your virtual environment ([Poetry installation](https://python-poetry.org/docs/#installation)) or [pip](https://pip.pypa.io/en/latest/development/), but feel free to use a different way of installing dependencies and managing virtual environments if you have other preferences.
+### Install dependencies and project
 
-#### Poetry
+Install the dependencies defined in `pyproject.toml` as well as the `federated-kaplan-meier-fitter` package.
```
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/README.md:53]:
```suggestion
Install the dependencies defined in `pyproject.toml` as well as the `examplefmk` package.
```

File: examples/federated-kaplan-meier-fitter/README.md [hunk 2]
```diff
@@ -25,86 +25,63 @@ the group it comes from therefore to simulate the division that might occur.
...
 
-You will see that the server is printing survival function, median survival time and saves the plot with the survival function.
-
-You can also check that the results match the centralized version.
-
-```shell
-$ python3 centralized.py
-```
+This example is a federated version of the centralized case that can be found
+[here](https://github.com/ml-explore/mlx-examples/tree/main/mnist).
```
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/README.md:87]:
```suggestion
```

File: examples/federated-kaplan-meier-fitter/README.md [hunk 3]
```diff
@@ -25,86 +25,63 @@ the group it comes from therefore to simulate the division that might occur.
...
+### Run with the Deployment Engine
 
-With both the long-running server (SuperLink) and two clients (SuperNode) up and running, we can now run the actual Flower App:
+> \[!NOTE\]
+> An update to this example will show how to run this Flower application with the Deployment Engine and TLS certificates, or with Docker.
 
-```bash
-flower-server-app server:app --insecure
-```
+## Explanation
```
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/README.md:84]:
```suggestion
```

File: examples/federated-kaplan-meier-fitter/examplefkm/task.py
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/examplefkm/task.py:1]: i think it's fine not to have a `task.py` just for this? wdyt?

File: examples/federated-kaplan-meier-fitter/examplefkm/task.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from lifelines.datasets import load_waltons
+
+from flwr_datasets.partitioner import NaturalIdPartitioner
+from datasets import Dataset
+
+
+def load_partition(partition_id: int):
+    X = load_waltons()
```
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/examplefkm/task.py:8]: would it make sense to have this outside this function acting like the _caching_ we have implemented for the templates?

File: examples/federated-kaplan-meier-fitter/README.md
Comment (jafermarq) [inline examples/federated-kaplan-meier-fitter/README.md:1]: Need to add `title` to the metadata section. This title should be the same as in the pyprojec.toml. Which should also match with the top title in the README.md

Comment (adam-narozniak) [reply examples/federated-kaplan-meier-fitter/README.md:1]: Solved: `Federated Survival Analysis with Flower and KaplanMeierFitter`

File: examples/federated-kaplan-meier-fitter/examplefkm/task.py (hunk 4 above)
Comment (adam-narozniak) [reply examples/federated-kaplan-meier-fitter/examplefkm/task.py:8]: I've moved the X outside of the function now; I guess that would be the best solution.

File: examples/federated-kaplan-meier-fitter/examplefkm/task.py
Comment (adam-narozniak) [reply examples/federated-kaplan-meier-fitter/examplefkm/task.py:1]: From my POV it's better to keep this structure even for a single function. But if you want me to change it it's also ok.

File: examples/federated-kaplan-meier-fitter/README.md (hunk 2 above)
Comment (adam-narozniak) [reply examples/federated-kaplan-meier-fitter/README.md]: Done

File: examples/federated-kaplan-meier-fitter/README.md (hunk 3 above)
Comment (adam-narozniak) [reply examples/federated-kaplan-meier-fitter/README.md]: Done

File: examples/federated-kaplan-meier-fitter/examplefkm/task.py
Comment (jafermarq) [reply examples/federated-kaplan-meier-fitter/examplefkm/task.py:1]: yes. let's keep it 👍

**Review by jafermarq — APPROVED** [review]

**Pull request merged**